### ISSUE TITLE
Support user provided schema

### ISSUE THREAD

**Issue by zhitaoli** [tensorflow/tfx#571]
In many cases, a schema file will be provided by the user. It's often the case the schema file would be created by SchemaGen, but then gets additional modification and blessing from user. The pipeline would use this user provided schema from that point.

**Comment by alionkun** [comment]
Add something about ExampleValidator component.
When the schema becomes steady, we don't need to use SchemaGen to build a new one, instead we use ExampleValidator to validate both 'eval' and 'train' split against the steady 'schema', this is the typical user case.
but ExampleValidator only validate the 'eval' split and other splits are ignored for now.
ExampleValidator need to support 'train' split.

**Comment by zhitaoli** [comment]
@alionkun yes your observation is correct and we should do that. I'll keep that as part of this issue.

**Comment by ucdmkt** [comment]
Here is a brief plan to address unlock this scenario.

1.  SchemaGen to optionally accept a fixed `Schema` artifact, which may be user-provided. If it is provided, SchemaGen passes through it. By doing it, schema is a tracked artifact in MLMD in the same way regardless of inferred or user-given, while SchemaGen's exec_properties differentiates how it was made.
2. A mechanism to 'import' a user-implemented schema.pbtxt in external uri as a schema artifact.
3. CLI command to 'export' current schema from pipeline to external uri (e.g. current directory) , regardless of inferred or user provided.
4. Make change to `ExampleValidator` to validate all splits against the provided schema (inferred or user-provided).

A PR to #1 is underway. #2 and #3 are under implementation.

**Comment by ucdmkt** [comment]
#577

**Comment by ucdmkt** [comment]
For (4), I filed a separate issue #628 because it can be done independently, not blocked on (2) and (3)

**Comment by andrewsmartin** [comment]
This should be handled now with the new [importer node](https://github.com/tensorflow/tfx/blob/master/tfx/components/common_nodes/importer_node.py) right?

**Comment by ruoyu90** [comment]
@andrewsmartin Yes! We just added an [example](https://github.com/tensorflow/tfx/blob/master/tfx/examples/chicago_taxi_pipeline/taxi_pipeline_importer.py) to demonstrate that.
